cmd/bucket-notification-handlers.go

```diff
@@ -250,10 +250,20 @@ func (api objectAPIHandlers) ListenBucketNotificationHandler(w http.ResponseWrit
 	bucket := vars["bucket"]
 
 	// Parse listen bucket notification resources.
-	prefix, suffix, events := getListenBucketNotificationResources(r.URL.Query())
-	if !IsValidObjectPrefix(prefix) || !IsValidObjectPrefix(suffix) {
-		writeErrorResponse(w, r, ErrFilterValueInvalid, r.URL.Path)
-		return
+	prefixes, suffixes, events := getListenBucketNotificationResources(r.URL.Query())
+
+	for _, prefix := range prefixes {
+		if !IsValidObjectPrefix(prefix) {
+			writeErrorResponse(w, r, ErrFilterValueInvalid, r.URL.Path)
+			return
+		}
+	}
+
+	for _, suffix := range suffixes {
```

Comment:
We could move this into a single function. Which validates and returns 'true/false' like following. 

```
validateFilterValues(values []string) APIErrorCode {
      for _, value := range values {
               if !IsValidObjectPrefix(value) {
                         return ErrFilterValueInvalid
               } 
      }
      return ErrNone
}
```

Now you can use this cleanly 

```
if apiErr := validateFilterValues(prefixes); apiErr != ErrNone {
              writeErrorResponse(w, r, apiErr, r.URL.Path)
}
```

This way you can add test cases for "validateFilterValues()"